src/renderer/components/TaskProgress.ts

```diff
@@ -0,0 +1,116 @@
+import { Task, TaskRuntime, TaskHandle } from "@xmcl/task";
+
+import taskProgressTemplate from "../templates/TaskProgress.pug";
+import { ResolvedVersion } from "@xmcl/core";
+
+export default class TaskProgress extends HTMLDivElement {
+	private $progress = () => this.getElementsByClassName("ui progress")[0];
+
+	private renderedTask: Task<ResolvedVersion> | null = null;
+	private taskCount: number = 0;
+	private incrementTaskCount(): void {
+		this.taskCount++;
+		if (this.taskCount === 1) {
+			// only 1 task in queue means there were no tasks before
+			$(this).transition("fly up in");
+		}
+	}
+	private decrementTaskCount(): void {
+		this.taskCount--;
+		if (this.taskCount === 0) {
+			// hide progress bar
+			$(this).transition("fly up out");
+		}
+	}
+
+	public constructor() {
+		super();
+	}
+
+	public connectedCallback(): void {
+		if (!this.hasChildNodes())
+			this.render();
+	}
+
+	public render(): void {
+		this.innerHTML = taskProgressTemplate();
+		$(this.$progress()).progress();
+		this.style.visibility = "hidden";
+	}
+
+	public addInstallTask(task: Task<ResolvedVersion>, instanceName: string): TaskRuntime<Task.State> {
+		const runtime = Task.createRuntime();
+		let rootNode: Task.State;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['rootNode' is assigned a value but never used.](https://app.codacy.com/manual/lukechu10/Minecraft-Box-Launcher/pullRequest?prid=5247069)